kernel/tests/log_compaction.rs

```diff
@@ -0,0 +1,209 @@
+use std::sync::Arc;
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use delta_kernel::engine::to_json_bytes;
+use delta_kernel::schema::{DataType, StructField, StructType};
+use delta_kernel::Snapshot;
+use object_store::path::Path;
+use object_store::ObjectStore;
+use test_utils::{create_table, engine_store_setup};
+use url::Url;
+
+/// Convert a URL to an object_store::Path
+fn url_to_object_store_path(url: &Url) -> Result<Path, Box<dyn std::error::Error>> {
+    let path_segments = url
+        .path_segments()
+        .ok_or_else(|| format!("URL has no path segments: {}", url))?;
+
+    let path_string = path_segments.skip(1).collect::<Vec<_>>().join("/");
+
+    Ok(Path::from(path_string))
+}
+
+#[tokio::test]
+async fn action_reconciliation_round_trip() -> Result<(), Box<dyn std::error::Error>> {
+    let _ = tracing_subscriber::fmt::try_init();
+
+    // Create a simple table schema: one int column named 'id'
+    let schema = Arc::new(StructType::try_new(vec![StructField::nullable(
+        "id",
+        DataType::INTEGER,
+    )])?);
+
+    // Setup engine and storage - this creates a proper temporary table
+    let (store, engine, table_location) = engine_store_setup("test_compaction_table", None);
+
+    // Create table (this will be commit 0)
+    let table_url = create_table(
+        store.clone(),
+        table_location,
+        schema.clone(),
+        &[],
+        false,
+        vec![],
+        vec![],
+    )
+    .await?;
+
+    // Commit 1: Add two files
+    let commit1_content = r#"{"add":{"path":"part-00000-file1.parquet","partitionValues":{},"size":1024,"modificationTime":1587968586000,"dataChange":true, "stats":"{\"numRecords\":10,\"nullCount\":{\"id\":0},\"minValues\":{\"id\": 1},\"maxValues\":{\"id\":10}}"}}
+{"add":{"path":"part-00001-file2.parquet","partitionValues":{},"size":2048,"modificationTime":1587968586000,"dataChange":true, "stats":"{\"numRecords\":20,\"nullCount\":{\"id\":0},\"minValues\":{\"id\": 11},\"maxValues\":{\"id\":30}}"}}
+{"commitInfo":{"timestamp":1587968586000,"operation":"WRITE","operationParameters":{"mode":"Append"},"isBlindAppend":true}}
```

Comment:
maybe just so we don't end up breaking stuff can we make `commitInfo` the first action in commit? this is required for ICT, etc. and kernel always writes it first.